tests/transactional_system/transactional_update.pm

```diff
@@ -61,7 +63,23 @@ sub run {
 
     record_info 'Install ptf', 'Install package - snapshot #1';
     trup_call "ptf install" . rpmver('security');
-    check_reboot_changes;
+    if (check_reboot_changes) {
+        # Reboot into new snapshot
+        power_action('reboot', observe => 0, keepconsole => 1);
+        # No grub bootloader on xen-pv
```

Comment:
https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/6391#discussion_r241006161

Reply:
Hi @kravciak, thanks for taking a look.

**transactional-update** is just the motivation to do that, since I am having "pain" adapting it for Tumbleweed and Leap.

Tumbleweed, Leap and SLE have an unified process of doing it that basically consist of calling **power_action** followed by **wait_boot**.

Also they handle grub menu, booting and login with test modules **grub_test**, **first_boot** (or **boot_to_desktop**) and then a module that does the login.

Do you mean that you are against unifying the current Tumbleweed,Leap,SLE with Kubic/CaaSP?

Reply:
ah, and the real tricky part is that wait_boot is an opensusebase test method, so it needs to be executed from the instance that inherited it. And in Kubic this is done by process_reboot which is a function on a library. I cannot call wait_boot from process_reboot without making a complicated fancy coding. So, I decided to do the opposite, bringing the logic from Kubic to the current TW,Leap,SLE.

And this is also part of the reason why, to avoid this problem in the future if/when we add more products/projects, I opened this discussion: https://progress.opensuse.org/issues/46589

Reply:
What do you mean by unifying? You want to replace `process_reboot` and `microos_login` by `wait_boot` sub? I think it's "it will break things" teritory.

I would call reboot/login until you have some transactional-update test running and then start refactoring reboot/login code.

Reply:
Yes, I mean replacing **process_reboot** and **microos_login** by **wait_boot**.

Well, you put me in a chicken or egg situation. I am exactly trying to have transactional-update scenario running for TW,Leap,SLE with transactional server role:

- https://openqa.opensuse.org/tests/latest?machine=64bit&flavor=DVD&arch=x86_64&test=transactional_host_install_only&version=Tumbleweed&distri=opensuse#
- https://openqa.suse.de/tests/latest?machine=aarch64&flavor=Installer-DVD&test=transactional_host_install_only&arch=aarch64&version=15-SP1&distri=sle
- https://openqa.suse.de/tests/latest?test=transactional_host_install_only&arch=ppc64le&machine=ppc64le&flavor=Installer-DVD&version=15-SP1&distri=sle
- https://openqa.suse.de/tests/latest?test=transactional_host_install_only&arch=s390x&machine=s390x-kvm-sle12&flavor=Installer-DVD&version=15-SP1&distri=sle#settings
- https://openqa.suse.de/tests/latest?flavor=Installer-DVD&machine=64bit&arch=x86_64&test=transactional_host_install_only&distri=sle&version=15-SP1

Now I need a working verification run for after installation modules, which I am trying to have on #6348 but, I created this PR (#6626) to make this controversial change separate.